src/sentry/profiles/task.py

```diff
@@ -308,22 +305,28 @@ def _get_event_instance(profile: Profile) -> Any:
         "os_name": profile["device_os_name"],
         "os_version": profile["device_os_version"],
         "retention_days": profile["retention_days"],
-        "call_trees": _get_call_trees(profile),
+        "call_trees": profile["call_trees"],
     }
 
 
-def _get_call_trees(profile: Profile) -> CallTrees:
+def _insert_vroom_profile(profile: Profile) -> bool:
     profile = dict(profile)
+    original_timestamp = profile["received"]
 
-    profile["received"] = (
-        datetime.utcfromtimestamp(profile["received"]).replace(tzinfo=timezone.utc).isoformat()
-    )
-
-    response = get_from_profiling_service(method="POST", path="/call_tree", json_data=profile)
-
-    # something went wrong, return empty call trees
-    if response.status != 200:
-        metrics.incr("profiling.get_call_tree", tags={"platform": profile["platform"]})
-        return {}
+    try:
+        profile["received"] = (
+            datetime.utcfromtimestamp(profile["received"]).replace(tzinfo=timezone.utc).isoformat()
+        )
+        response = get_from_profiling_service(method="POST", path="/profile", json_data=profile)
 
-    return cast(CallTrees, json.loads(response.data)["call_trees"])
+        if response.status == 204:
+            profile["call_trees"] = {}
+        elif response.status == 200:
+            profile["call_trees"] = json.loads(response.data)["call_trees"]
+        else:
+            metrics.incr("profiling.insert_vroom_profile", tags={"platform": profile["platform"]})
+            return False
+        return True
+    finally:
+        profile["received"] = original_timestamp
+        profile["profile"] = ""
```

Comment:
do we want to keep writing to clickhouse temporarily to make sure that the gcs data is good?

Reply:
`vroom` PR related to this is merged and already writing to GCS. We can check the data is good and then merge this, no need to split this PR.